UI lightbox: Replace '<', '>' and 'X' with icons.

jitendra-ky opened the pull request:
<!-- Describe your pull request here.-->
This pull request updates the lightbox overlay by replacing the '<', '>', and 'X' characters with icons for a better visual appearance. Additionally, the icons are now center-aligned.

Fixes: #33068 

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
---
|   Font size  | **after** | **before** |
| -- | -- | -- |
| 12px| ![image](https://github.com/user-attachments/assets/01d48320-e34d-418e-8a62-8d186147568e) ![image](https://github.com/user-attachments/assets/7863270d-14f4-462a-a043-517bec242da6) |  ![image](https://github.com/user-attachments/assets/c1ac40d0-8b1b-4587-b097-a92b12182a61) ![image](https://github.com/user-attachments/assets/5fae27d3-27bf-4860-9841-4cd316916e8b)|
| 20px | ![image](https://github.com/user-attachments/assets/6ecf0ca0-c15b-4d64-8c93-699bdd9ada39) ![image](https://github.com/user-attachments/assets/a2958e47-98c9-4679-bfe1-e2009515b8c2) | ![image](https://github.com/user-attachments/assets/53f8356c-1464-41df-b89a-5df8eb33438f)  ![image](https://github.com/user-attachments/assets/4cf47da7-ff86-4292-8722-bc6a491fc6ca)|

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
@roanster007 would you be up for reviewing this one?

roanster007 commented on web/templates/lightbox_overlay.hbs:
```diff
@@ -18,8 +22,8 @@
+        <div class="arrow no-select" data-direction="prev"><i class="zulip-icon zulip-icon-chevron-left"></i></div>
         <div class="image-list"></div>
-        <div class="arrow no-select" data-direction="next">&gt;</div>
+        <div class="arrow no-select" data-direction="next"><i class="zulip-icon zulip-icon-chevron-right"></i></div>
```
Why did you prefer an `<i>` over `<span>` tag, considering `<span>` is inline too?

roanster007 commented:
@jitendra-ky  changes look good to me, apart from the density of the left and right icons compared to current ones:

Before:
![image](https://github.com/user-attachments/assets/822a1696-4385-49fa-bbe9-4a073c52e452)

After:
![image](https://github.com/user-attachments/assets/30c5ffba-57d5-4f92-9053-b3edb2e38f9e)

If you feel they are fine, might be better to verify once in the #design thread.

jitendra-ky commented:
> If you feel they are fine, might be better to confirm once in the #design thread.

created the thread: [CZO thread #design](https://chat.zulip.org/#narrow/channel/101-design/topic/lightbox.20icons.20.22.3C.22.20and.20.22.3E.22/with/2146552)

jitendra-ky commented:
Updated the `aero` style to maintain a consistent spacing from the image — same as before.  
Also updated the screenshots.

@roanster007,  how does it look now?

roanster007 commented:
@jitendra-ky Looks good to me!

alya commented:
@karlstolley Any feedback on this one? Please tag for integration review if it looks good from your perspective.

karlstolley commented on web/styles/lightbox.css:
```diff
@@ -50,19 +50,23 @@
         cursor: pointer;
         transition: opacity 0.2s ease;
+
+        position: relative;
```
I would much prefer a flexbox rewrite in this area and not use positioning like this. 

I also don't understand the use of `rem` units on the height and width here. Shouldn't this scale with everything else?

karlstolley commented on web/styles/lightbox.css:
```diff
@@ -180,21 +184,31 @@
-            line-height: 1.6rem;
-            font-weight: 200;
+            height: 50px;
+            width: 0.7em; /* 14 px at 20px/em */
```
Why is the width set in ems, but the height set in pixels?

jitendra-ky replied:
height is 50px to match the height of image, and the width set to 0.7em as it need to scale with icon. (to keep same as before)

## at 16px
![image](https://github.com/user-attachments/assets/28ce51be-f6c2-43a0-a49b-a06faab6ab8a)
## at 20px
![image](https://github.com/user-attachments/assets/80199d21-b429-4edd-acae-9acb21a7cd08)

karlstolley commented on web/styles/lightbox.css:
```diff
@@ -180,21 +184,31 @@
+                position: absolute;
+                top: 50%;
+                left: 50%;
+                transform: translate(-50%, -50%) scale(2);
```
Why does this instance also have scale()?

timabbott commented:
Closing as this was completed via #33068. Thanks for the effort @jitendra-ky.
